backend/src/main/java/com/redhat/cloud/notifications/db/repositories/DrawerNotificationRepository.java

```diff
@@ -40,10 +40,9 @@ public List<DrawerEntryPayload> getNotifications(String orgId, String username,
         query.setSortFields(DrawerNotification.SORT_FIELDS);
         query.setDefaultSortBy("created:DESC");
         Optional<Query.Sort> sort = query.getSort();
-
         String hql = "SELECT dn.id, dn.read, " +
-            "dn.event.bundleDisplayName, dn.event.applicationDisplayName, dn.event.eventTypeDisplayName, dn.created, dn.event.renderedDrawerNotification "
-            + "FROM DrawerNotification dn where dn.orgId = :orgId and dn.userId = :userid";
+            "ev.bundleDisplayName, ev.applicationDisplayName, ev.eventTypeDisplayName, dn.created, ev.renderedDrawerNotification "
+            + "FROM DrawerNotification dn inner join dn.event as ev where dn.orgId = :orgId and dn.userId = :userid";
```

Comment:
Does that `inner join` have any impact on the response time? In the old code, Hibernate took care of that join automatically, maybe in a slightly different way.

Reply:
I'm very surprised to hear that Hibernate would do a cross join because it makes no sense based on the HQL query and entities mapping. Maybe there was a bug in Hibernate 5...

I just checked with Hibernate 6 and the SQL contains a default `join` (which is inner).

Reply:
Indeed generated SQL request seems have changed :)